src/DaemonExec.ts

```diff
@@ -0,0 +1,190 @@
+import * as Docker from 'dockerode'
+
+interface ContainerModem {
+	demuxStream(
+		stream: NodeJS.ReadWriteStream,
+		stdoutStream: NodeJS.WritableStream,
+		stderrStream: NodeJS.WritableStream,
+	): void
+}
+
+interface DockerModem {
+	followProgress(
+		stream: NodeJS.ReadableStream,
+		onFinished: (err?: Error) => Promise<void>,
+		onProgress: (event: {
+			status: string
+			progressDetail: { current: number; total: number }
+			progress: string
+			id: string
+		}) => void,
+	): void
+}
+
+export interface VolumeDefinition {
+	fromPath: string
+	toPath: string
+}
+
+/**
+ * Returns information about the images
+ * @param docker - The docker daemon to pull information from
+ * @returns an array of ImageInfo
+ */
+export async function listImages(docker: Docker): Promise<Docker.ImageInfo[]> {
+	const images = await docker.listImages({})
+	return images
+}
+
+/**
+ * Returns information about the containers
+ * @param docker - The docker daemon to pull information from
+ * @returns an array of ContainerInfo
+ */
+export async function listContainers(
+	docker: Docker,
+): Promise<Docker.ContainerInfo[]> {
+	const containers = await docker.listContainers()
+	return containers
+}
+
+/**
+ * Starts the import of an image
+ * @param docker - The docker daemon with which the image will be stored
+ * @param name - The name of the image to pull
+ * @returns a ReadableStream to track the progress of the import
+ */
+export async function importImage(
+	docker: Docker,
+	name: string,
+): Promise<NodeJS.ReadWriteStream> {
+	const stream: NodeJS.ReadWriteStream = await (docker.pull(
+		name,
+	) as Promise<NodeJS.ReadWriteStream>)
```

Comment:
Why do you need to use `as` here?

Reply:
In the definition for docker.pull, it returns an any type, so we have to use "as" to assign the output to a NodeJS.ReadWriteStream.